alchemist-incarnation-collektive/src/main/kotlin/it/unibo/alchemist/CollektiveIncarnation.kt

```diff
@@ -90,7 +111,37 @@ class CollektiveIncarnation<P> : Incarnation<Any?, P> where P : Position<P> {
         randomGenerator: RandomGenerator,
         environment: Environment<Any?, P>,
         parameter: String?,
-    ): Node<Any?> = GenericNode(environment).also {
-        it.addProperty(CollektiveDevice(environment, it, DoubleTime(parameter?.toDoubleOrNull() ?: 0.0)))
+    ): Node<Any?> = GenericNode(environment).also { genericNode ->
+        genericNode.addProperty(
+            CollektiveDevice(
+                environment,
+                genericNode,
+                parameter?.toDoubleOrNull().let {
+                    if (it != null) DoubleTime(it) else null
+                },
```

Comment:
you are checking for nullability too many times :smiley: 
I was about to suggest, but I think you should improve these three lines yourself, do it and show me the result.

Reply:
I improved it like this
```suggestion
               parameter?.let { DoubleTime(it.toDouble()) },
```